### PR TITLE
[bitnami/harbor-notary-signer] Add VIB tests

### PR DESCRIPTION
**Pull request by corico44**
### Description of the change

The main objective of this PR is to publish our Bitnami Harbor-notary-signer container using VMware Image Builder. In order to do that, several changes are included:

- Increasing the existing test coverage of the asset by adding Goss tests.
- Update verify and publish VIB pipeline's definitions.

### Benefits

- Ensuring higher quality of the container catalog.
- Increased pool of assets completely handled by VMware Image Builder.

### Possible drawbacks

Automated tests could introduce additional flakiness to the CI/CD.

### Applicable issues

NA

### CODE REVIEW

File: .vib/harbor-notary-signer/goss/harbor-notary-signer.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+file:
+  /etc/notary:
+    exists: true
+    mode: "0775"
+    owner: harbor
+    filetype: directory
+command:
+  check-app-run:
+    exec: notary-signer --help
+    exit-status: 2
```
Comment (migruiz4) [inline .vib/harbor-notary-signer/goss/harbor-notary-signer.yaml:16]: Please replace the help test with a version test.

Version can be displayed by running the `notary-signer` command without flags, although exit-status will return 1.

```
I have no name!@0c2cf4a39717:/vib$ notary-signer       
{"level":"info","msg":"Version: 0.6.1, Git commit: d6e1431f","time":"2023-04-25T09:42:31Z"}
{"level":"fatal","msg":"Could not read config at :, viper error: Unsupported Config Type \"\"","time":"2023-04-25T09:42:31Z"}
I have no name!@0c2cf4a39717:/vib$ echo $?
1
```

Comment (corico44) [reply .vib/harbor-notary-signer/goss/harbor-notary-signer.yaml]: When executing the notary-signer command, we can see that the version that is being displayed isn't the same version as the version of the application.
```
HOSTNAME=a2c3cd4dfab6
PWD=/tmp
OS_FLAVOUR=debian-11
HOME=/
SERVICE_NAME=notary_signer
TERM=xterm
SHLVL=1
BITNAMI_APP_NAME=harbor-notary-signer
APP_VERSION=2.8.0
OS_NAME=linux
PATH=/opt/bitnami/harbor-notary-signer/bin:/opt/bitnami/common/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
OS_ARCH=amd64
_=/usr/bin/env
OLDPWD=/tmp/bitnami

root@a2c3cd4dfab6:/tmp# notary-signer
{"level":"info","msg":"Version: 0.6.1, Git commit: d6e1431f","time":"2023-04-27T11:49:56Z"}
{"level":"fatal","msg":"Could not read config at :, viper error: Unsupported Config Type \"\"","time":"2023-04-27T11:49:56Z"}
```

File: .vib/harbor-notary-signer/goss/harbor-notary-signer.yaml
Comment (FraPazGal) [inline .vib/harbor-notary-signer/goss/harbor-notary-signer.yaml:1]: Unless I'm mistaken, you missed adding the checks related to the changes performed for internal TLS.

File: .vib/harbor-notary-signer/goss/vars.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+binaries:
+  - notary-signer
```
Comment (FraPazGal) [inline .vib/harbor-notary-signer/goss/vars.yaml:2]: As done for `harbor-notary-server`, we should also check the `migrate` binary exists.

**Review by FraPazGal — COMMENTED** [review]
Nice PR @corico44! I just added a couple of comments as a review.

File: .vib/harbor-notary-signer/goss/harbor-notary-signer.yaml
Comment (corico44) [reply .vib/harbor-notary-signer/goss/harbor-notary-signer.yaml:1]: I don't see the check being done in any file.

Comment (FraPazGal) [reply .vib/harbor-notary-signer/goss/harbor-notary-signer.yaml:1]: You are right sorry, I mixed the assets and ended up looking a different postunpack logic. 😓

**Review by FraPazGal — APPROVED** [review]
Thanks for the PR! LGTM!

**Pull request merged**